### PR TITLE
Added OpenAPI docs for V2 API Specification

### LINKED ISSUE #2308: Add OpenAPI Docs for V2 API Specification
As discussed during the Core Working Group meeting 16-Jan-2020 the OpenAPI specification documents which have so far been developing in my [personal repo](https://github.com/tsconn23/edgex-geneva-api) need to be added to the edgex-go repository under an api/openapi directory.

### PR DESCRIPTION
**Pull request by tsconn23**
Fix #2308

Signed-off-by: Trevor Conn <trevor_conn@dell.com>

### CODE REVIEW

**Comment by tsconn23** [comment]
@michaelestrin Per your review comments above and our chat earlier this afternoon, I have pushed individual commits to address points 1, 3, 5 and 7.

During our conversation this afternoon, we deferred 2 agreeing that the endpoints would use the specific use-case response type making a "best effort" attempt to populate it.

For 4 we agreed further discussion w/r/t defining the mapping of HTTP response codes to transport-agnostic status codes was needed, and could be handled outside of this PR.

6 and 8 remain to be done. As an FYI, our GET endpoints actually do return DTOs but they don't have requests defined. See [DeviceResponse](https://github.com/tsconn23/edgex-go/blob/4ef9e0da681074669e017c9cb6f7fca88d16e6cf/api/openapi/core-metadata.yaml#L377) for example. More work remains to normalize the naming and to create the requisite request types. Will update when that work is done.

**Comment by cloudxxx8** [comment]
@tsconn23 we will also put open api files for V1 APIs.  could you please use folder or filename to differentiate them?  thanks

File: api/openapi/core-command.yaml [hunk 1]
```diff
@@ -61,34 +61,23 @@ components:
...
       type: object
       properties:
-        correlationId:
-          description: "Identifies the batch to which this request belongs. If used via REST, this value should be the same as that of  the ''X-Correlation-ID'' HTTP header."
-          type: string
-          format: uuid
-          example: "82eb2e26-0f24-48aa-ae4c-de9dac3fb9bc"
         requestId:
-          description: "Uniquely identifies this request apart from its peers in the batch. For implementation, recommend this value be generated by the type's constructor."
+          description: "Uniquely identifies this request apart from its peers in a batch. For implementation, recommend this value be generated by the type's constructor."
```
Comment (michaelestrin) [inline api/openapi/core-command.yaml]: `apart from its peers in a batch` should be removed.  Commented here; applies elsewhere.

File: api/openapi/core-command.yaml [hunk 2]
```diff
@@ -61,34 +61,23 @@ components:
...
       type: object
       properties:
-        correlationId:
-          description: "Identifies the batch to which this response belongs. If used via REST, this value should be the same as that of  the ''X-Correlation-ID'' HTTP header."
-          type: string
-          format: uuid
-          example: "82eb2e26-0f24-48aa-ae4c-de9dac3fb9bc"
         requestId:
-          description: "Uniquely identifies the request within the larger batch resulting in this response."
+          description: "Uniquely identifies the request within the larger batch that resulted in this response."
```
Comment (michaelestrin) [inline api/openapi/core-command.yaml]: `within the larger batch ` should be removed.  Commented here; applies elsewhere

File: api/openapi/core-metadata.yaml [hunk 3]
```diff
@@ -1078,7 +1078,9 @@ paths:
     parameters:
       - $ref: '#/components/parameters/correlatedRequestHeader'
     post:
-      summary: "Accepts an arbitrary number of RequestEnvelope instances. Each RequestEnvelope has a 'content' property that holds a specific request type. This endpoint is operation agnostic will route the request to the appropriate handler implementation within the service."
+      servers:
+      - url:  http://localhost:48081/api
```
Comment (michaelestrin) [inline api/openapi/core-metadata.yaml]: Not sure if this had the desired effect.  Pull open document in editor.swagger.io -- don't see /batch represented differently from other endpoints:

image

Commented here; applies elsewhere.

File: api/openapi/core-command.yaml [hunk 4]
```diff
@@ -137,6 +148,47 @@ components:
...
+        cpuUsedPercent:
+          description: "The average % of CPU utilization"
+          type: number
+        memoryUsed:
+          description: "The total bytes of memory obtained from the OS."
+          type: integer
+        raw:
+          description: "A marshalled response type containing more granular metrics."
+          type: string
+          format: byte
```
Comment (michaelestrin) [inline api/openapi/core-command.yaml]: This is accurate for the metrics endpoint on the system management service but not for the metrics endpoint on the core services.  Initial version of content looks like this:

```
{
  "memAlloc": 869640,
  "memTotalAlloc": 869640,
  "memSys": 6949240,
  "memMallocs": 7750,
  "memFrees": 531,
  "memLiveObjects": 7219,
  "cpuBusyAvg": 80.32117867226148
}
```

Commented here; applies elsewhere.

File: api/openapi/core-metadata.yaml (hunk 3 above)
Comment (tsconn23) [reply api/openapi/core-metadata.yaml]: Yeah, this is not easy to find in the UI. What you have to do is click on the endpoint to get the example request/response shown. You should then see a "Try it Out" button to the right, aligned with the "Parameters" tab. Click that and then scroll down until you see the blue "Execute" button. On top of that you will see the alternate server listed in the dropdown.
OpenAPI_Execute

File: api/openapi/core-command.yaml (hunk 1 above)
Comment (tsconn23) [reply api/openapi/core-command.yaml]: Addressed. Per Cloud's comment, YAML docs are now in a v2 folder.

File: api/openapi/core-command.yaml (hunk 2 above)
Comment (tsconn23) [reply api/openapi/core-command.yaml]: Addressed. Per Cloud's comment, YAML docs are now in a v2 folder.

File: api/openapi/core-command.yaml (hunk 4 above)
Comment (tsconn23) [reply api/openapi/core-command.yaml]: Addressed. Per Cloud's comment, YAML docs are now in a v2 folder.

**Pull request merged**